### PR TITLE
PWN-7351 - fix details and history

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7351

## Description of Work

PWN-7351 - fix details and items showing for burn and mint, and icons fixes

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/history/model/rpc/RpcHistoryTransaction.kt [hunk 1]
```diff
@@ -60,12 +60,11 @@ sealed class RpcHistoryTransaction(
...
         @DrawableRes
-        fun getIcon(): Int = if (isBurn) R.drawable.ic_transaction_send else R.drawable.ic_transaction_receive
+        fun getIcon(): Int = R.drawable.ic_renbtc
 
-        fun getValue(): String = "${getSymbol(isBurn)} ${getFormattedAmount()} ${Constants.USD_SYMBOL}"
+        fun getUsdAmount(): String = "${getFormattedAmount()} ${Constants.USD_SYMBOL}"
 
-        fun getTotal(): String =
-            "${getSymbol(isBurn)} ${amount.total.scaleMedium().formatToken()} ${Constants.REN_BTC_SYMBOL}"
+        fun getTotal(): String = "${amount.total.scaleMedium().formatToken()} ${Constants.REN_BTC_SYMBOL}"
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/model/rpc/RpcHistoryTransaction.kt:67]: decimals for formatToken?

File: app/src/main/java/org/p2p/wallet/history/model/rpc/RpcHistoryTransaction.kt [hunk 2]
```diff
@@ -74,6 +73,23 @@ sealed class RpcHistoryTransaction(
...
+            !status.isCompleted() -> {
+                R.color.text_rose
+            }
+            isBurn -> {
+                R.color.text_night
+            }
+            else -> {
+                R.color.text_mint
+            }
+        }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/model/rpc/RpcHistoryTransaction.kt:91]:
```suggestion
        fun getTextColor(): Int = when {
            !status.isCompleted() -> R.color.text_rose
            isBurn -> R.color.text_night
            else -> R.color.text_mint
        }
```

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsBottomSheetPresenter.kt [hunk 3]
```diff
@@ -197,10 +196,13 @@ class HistoryTransactionDetailsBottomSheetPresenter(
...
-                val isSend = transaction is RpcHistoryTransaction.Transfer && transaction.isSend
-                val notGreen = transaction is RpcHistoryTransaction.Swap || isSend
                 titleRes = R.string.transaction_details_title_succeeded
-                colorRes = if (notGreen) R.color.text_night else R.color.text_mint
+                colorRes = when {
+                    transaction is RpcHistoryTransaction.Transfer && transaction.isSend ||
+                        transaction is RpcHistoryTransaction.BurnOrMint && transaction.isBurn ||
+                        transaction is RpcHistoryTransaction.Swap -> R.color.text_night
+                    else -> R.color.text_mint
+                }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsBottomSheetPresenter.kt:205]:
```suggestion
val isTransactionTransfer = transaction is RpcHistoryTransaction.Transfer && transaction.isSend
                val isTransactionBurn = transaction is RpcHistoryTransaction.BurnOrMint && transaction.isBurn
                val isTransactionSwap = transaction is RpcHistoryTransaction.Swap
                colorRes = if (isTransactionTransfer || isTransactionBurn || isTransactionSwap) {
                    R.color.text_night
                } else {
                    R.color.text_mint
                }
```

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsBottomSheetPresenter.kt:205]: why do you need double check RpcHistoryTransaction.Transfer for transaction.isSend? 
i thought Transfer == Send

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/history/ui/detailsbottomsheet/HistoryTransactionDetailsBottomSheetPresenter.kt:205]: Transfer == Send/Receive

File: app/src/main/java/org/p2p/wallet/history/model/rpc/RpcHistoryTransaction.kt [hunk 4]
```diff
@@ -74,6 +73,17 @@ sealed class RpcHistoryTransaction(
...
+        fun getFormattedAbsTotal(): String {
+            return "${amount.total.abs().scaleLong().toPlainString()} ${Constants.REN_BTC_SYMBOL}"
+        }
+
+        @ColorRes
+        fun getTextColor(): Int = when {
+            !status.isCompleted() -> R.color.text_rose
+            isBurn -> R.color.text_night
+            else -> R.color.text_mint
+        }
```
Comment (NikolaiEmelianov) [inline app/src/main/java/org/p2p/wallet/history/model/rpc/RpcHistoryTransaction.kt:85]: bad practic to mix ui and domain

**Review by NikolaiEmelianov — APPROVED** [review]

**Pull request merged**